dkist/dataset/tiled_dataset.py

```diff
@@ -150,6 +128,18 @@ def __init__(
         self._data = np.ma.masked_array(dataset_array, dtype=object, mask=mask)
         meta = meta or {}
         inventory = meta.get("inventory", inventory or {})
+
+        # If headers are saved as one Table for the whole TiledDataset, use those first
+        # Otherwise stack the headers saved for component Datasets
+        if meta.get("headers") is None:
```

Comment:
I think this bit is missing an explicit test? It would be good to add one which makes sure that the dkist-inventory workflow behaves as expected.